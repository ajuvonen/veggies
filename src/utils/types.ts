import type {DateTime} from 'luxon';

export type Locale = 'en' | 'fi';

export enum Category {
  Fruit = 'Fruit',
  Vegetable = 'Vegetable',
  Leafy = 'Leafy',
  Root = 'Root',
  Bean = 'Bean',
  Grain = 'Grain',
  Mushroom = 'Mushroom',
}

export type Favorites = {
  [key in Category]: readonly [string, number][];
};

export type Week = {
  veggies: string[];
  readonly startDate: DateTime;
};

export type Challenge = {
  readonly startDate: DateTime;
  veggie: string | undefined;
};

export type Settings = {
  allergens: string[];
  locale: Locale;
  showChartAnimations: boolean;
  suggestionCount: number;
  summaryViewedDate: DateTime | null;
};

export type Listing = {
  readonly veggie: string;
  readonly category: Category;
};

export type TranslatedListing = Listing & {
  readonly translation: string;
  readonly synonyms: string[];
};

export enum AchievementLevel {
  NoAchievement,
  Bronze,
  Silver,
  Gold,
  Platinum,
}

export type Achievements = {
  allOnRed: AchievementLevel;
  botanicalBerries: AchievementLevel;
  challengeAccepted: AchievementLevel;
  committed: AchievementLevel;
  completionist: AchievementLevel;
  experimenterBean: AchievementLevel;
  experimenterFruit: AchievementLevel;
  experimenterGrain: AchievementLevel;
  experimenterLeafy: AchievementLevel;
  experimenterMushroom: AchievementLevel;
  experimenterRoot: AchievementLevel;
  experimenterVegetable: AchievementLevel;
  goNuts: AchievementLevel;
  hotStreak: AchievementLevel;
  lemons: AchievementLevel;
  thirtyVeggies: AchievementLevel;
  thousandsOdd: AchievementLevel;
  thousandsEven: AchievementLevel;
};

export type WeekData = {
  atMostVeggies: number;
  challenge: string | undefined;
  firstTimeVeggies: string[];
  firstWeek: boolean;
  hotStreak: number;
  mean: number;
  previousWeekCount: number;
  veggies: string[];
  weekNumber: string;
};

export type SummaryItem = {
  emoji: string;
  translationKey: string;
  translationParameters: (string | number)[];
};
