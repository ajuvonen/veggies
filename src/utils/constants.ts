import type {InjectionKey, Ref} from 'vue';
import type {UseMemoizeReturn} from '@vueuse/core';
import {Category, type Locale} from './types';

export const APP_URL = 'https://eatyourveggies.app';

export const BLUESKY_URL = 'https://bsky.app/profile/eatyourveggies.app';

export const PLAY_STORE_URL =
  'https://play.google.com/store/apps/details?id=app.eatyourveggies.twa';

export const BUILD_TIME = __VITE_BUILD_TIME__;

export const DEFAULT_LOCALE: Locale = 'en';

export const DEFAULT_SETTINGS = {
  allergens: [] as string[],
  locale: DEFAULT_LOCALE,
  showChartAnimations: true,
  suggestionCount: 10,
} as const;

export const KEYS = {
  challenge: Symbol() as InjectionKey<Ref<string | undefined>>,
  dropdownStyles: Symbol() as InjectionKey<
    UseMemoizeReturn<string, [active: boolean, selected: boolean]>
  >,
} as const;

export const LOCALES: readonly Locale[] = ['en', 'fi'] as const;

export const COLORS = {
  chartColors: ['#f0f9ff', '#bae6fd', '#38bdf8', '#0284c7', '#075985', '#082f49', '#0B1623'],
  chartColorsAlternate: [
    '#fdc5f5',
    '#f7aef8',
    '#b388eb',
    '#8093f1',
    '#79b8f4',
    '#72ddf7',
    '#bae6fd',
  ],
  offWhite: '#f9fafb',
  darkBlue: '#0c4a6e',
} as const;

export const CATEGORY_EMOJI = {
  [Category.Fruit]: 'üçé',
  [Category.Vegetable]: 'ü•¶',
  [Category.Leafy]: 'ü•¨',
  [Category.Root]: 'ü•ï',
  [Category.Bean]: 'ü´õ',
  [Category.Grain]: 'üåæ',
  [Category.Mushroom]: 'üçÑ‚Äçüü´',
} as const;

export const FRUITS: readonly string[] = [
  'apple',
  'apricot',
  'asian pear',
  'banana',
  'blackberry',
  'blackcurrant',
  'blood grapefruit',
  'blood orange',
  'blueberry',
  'boysenberry',
  'calamansi',
  'cantaloupe',
  'cape gooseberry',
  'caper',
  'cherimoya',
  'cherry',
  'clementine',
  'cloudberry',
  'coconut',
  'cranberry',
  'date',
  'dragon fruit',
  'durian',
  'fig',
  'finger lime',
  'gooseberry',
  'grape',
  'grapefruit',
  'grenadilla',
  'guava',
  'honeydew melon',
  'jackfruit',
  'kiwi',
  'kumquat',
  'lemon',
  'lime',
  'lingonberry',
  'longan',
  'lychee',
  'mandarin',
  'mango',
  'mangosteen',
  'nectarine',
  'orange',
  'papaya',
  'passion fruit',
  'peach',
  'pear',
  'persimmon',
  'piel de sapo',
  'pineapple',
  'plum',
  'pomegranate',
  'pomelo',
  'prickly pear',
  'quince',
  'rambutan',
  'raspberry',
  'redcurrant',
  'sea buckthorn',
  'starfruit',
  'strawberry',
  'watermelon',
];

export const LEAFIES: readonly string[] = [
  'arugula',
  'basil',
  'bok choy',
  'cabbage',
  'chard',
  'choy sum',
  'cilantro',
  'collard greens',
  'dandelion greens',
  'dill',
  'endive',
  'frisee',
  'holy basil',
  'iceberg lettuce',
  'kale',
  'lettuce',
  'little gem lettuce',
  'mache',
  'mint',
  'mizuna',
  'mustard greens',
  'napa cabbage',
  'oak leaf lettuce',
  'parsley',
  'pea sprouts',
  'radicchio',
  'red cabbage',
  'red leaf lettuce',
  'romaine',
  'savoy cabbage',
  'sorrel',
  'spinach',
  'sprouts',
  'thai basil',
  'thyme',
  'water spinach',
  'watercress',
];

export const ROOTS: readonly string[] = [
  'bamboo shoots',
  'beet',
  'black radish',
  'black salsify',
  'carrot',
  'celeriac',
  'chives',
  'daikon',
  'fennel',
  'garlic',
  'ginger',
  'horseradish',
  'jerusalem artichoke',
  'jicama',
  'kohlrabi',
  'leek',
  'manioc',
  'onion',
  'parsley root',
  'parsnip',
  'potato',
  'radish',
  'red onion',
  'rutabaga',
  'scallion',
  'shallot',
  'sweet potato',
  'taro',
  'turnip',
  'wasabi',
  'white onion',
  'yam',
  'yellow beet',
];

export const VEGETABLES: readonly string[] = [
  'artichoke',
  'asparagus',
  'avocado',
  'bell pepper',
  'bitter melon',
  'broccoli',
  'broccolini',
  'brussels sprouts',
  'butternut squash',
  'cauliflower',
  'celery',
  'chayote',
  'cherry tomato',
  'chili pepper',
  'corn',
  'cucumber',
  'eggplant',
  'hokkaido pumpkin',
  'jalapeno',
  'nori',
  'okra',
  'olive',
  'plantain',
  'pointed pepper',
  'pumpkin',
  'rhubarb',
  'spaghetti squash',
  'tomatillo',
  'tomato',
  'wakame',
  'wax gourd',
  'zucchini',
];

export const MUSHROOMS: readonly string[] = [
  'black trumpet mushroom',
  'champignon',
  'chanterelle',
  'enoki',
  'hedgehog mushroom',
  'king oyster mushroom',
  'matsutake',
  'morel',
  'oyster mushroom',
  'porcini',
  'portobello',
  'shiitake',
  'shimeji',
  'wild mushroom',
  'wood ear mushroom',
  'yellowfoot mushroom',
];

export const BEANS: readonly string[] = [
  'black bean',
  'black lentil',
  'black-eyed pea',
  'brown lentil',
  'butter bean',
  'cannellini bean',
  'chickpea',
  'cranberry bean',
  'edamame',
  'fava bean',
  'flageolet bean',
  'green bean',
  'green lentil',
  'kidney bean',
  'mung bean',
  'pea',
  'pinto bean',
  'red bean',
  'red lentil',
  'scarlet runner bean',
  'snow pea',
  'sugar snap pea',
  'soybean',
  'white bean',
  'yardlong bean',
  'yellow lentil',
];

export const GRAINS: readonly string[] = [
  'almond',
  'amaranth',
  'barley',
  'brazil nut',
  'buckwheat',
  'bulgur',
  'cashew nut',
  'chestnut',
  'chia seed',
  'couscous',
  'farro',
  'flaxseed',
  'hazelnut',
  'hemp seed',
  'kamut',
  'macadamia nut',
  'millet',
  'oat',
  'peanut',
  'pecan nut',
  'pine nut',
  'pistachio nut',
  'poppy seed',
  'pumpkin seed',
  'quinoa',
  'rice',
  'rye',
  'safflower seed',
  'sesame seed',
  'sorghum',
  'spelt',
  'sunflower seed',
  'teff',
  'triticale',
  'walnut',
  'wheat',
  'wild rice',
];

export const ALL_VEGGIES: readonly string[] = [
  ...FRUITS,
  ...VEGETABLES,
  ...LEAFIES,
  ...ROOTS,
  ...BEANS,
  ...GRAINS,
  ...MUSHROOMS,
];

export const CITRUSES: readonly string[] = [
  'blood grapefruit',
  'blood orange',
  'calamansi',
  'clementine',
  'finger lime',
  'grapefruit',
  'kumquat',
  'lemon',
  'lime',
  'mandarin',
  'orange',
  'pomelo',
];

export const RED_VEGGIES: readonly string[] = [
  'apple',
  'beet',
  'bell pepper',
  'blood grapefruit',
  'blood orange',
  'cherry tomato',
  'cherry',
  'chili pepper',
  'cranberry',
  'lingonberry',
  'pointed pepper',
  'pomegranate',
  'radish',
  'raspberry',
  'red bean',
  'red cabbage',
  'red onion',
  'redcurrant',
  'strawberry',
  'tomato',
  'watermelon',
];

export const NUTS: readonly string[] = [
  'almond',
  'brazil nut',
  'cashew nut',
  'chestnut',
  'coconut',
  'hazelnut',
  'macadamia nut',
  'peanut',
  'pecan nut',
  'pine nut',
  'pistachio nut',
  'walnut',
];

export const BOTANICAL_BERRIES: readonly string[] = [
  'banana',
  'bell pepper',
  'blood grapefruit',
  'blood orange',
  'blueberry',
  'cantaloupe',
  'cape gooseberry',
  'cherry tomato',
  'chili pepper',
  'clementine',
  'cranberry',
  'cucumber',
  'eggplant',
  'grape',
  'grapefruit',
  'guava',
  'honeydew melon',
  'kiwi',
  'kumquat',
  'lemon',
  'lime',
  'mandarin',
  'orange',
  'papaya',
  'passion fruit',
  'piel de sapo',
  'plantain',
  'pointed pepper',
  'pomelo',
  'pumpkin',
  'tomatillo',
  'tomato',
  'watermelon',
  'zucchini',
];
