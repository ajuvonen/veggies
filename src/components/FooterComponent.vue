<script lang="ts" setup>
import {computed} from 'vue';
import {BLUESKY_URL, PLAY_STORE_URL} from '@/utils/constants';

const showPlayStoreLink = computed(
  () => navigator.userAgent.includes('Android') && !navigator.userAgent.includes('wv'),
);
</script>
<template>
  <footer>
    <a
      class="flex-container items-center"
      :href="BLUESKY_URL"
      target="_blank"
      noreferrer
      noopener
      data-test-id="bluesky-link"
    >
      <IconComponent icon="bluesky" />
      {{ $t('general.follow') }}
    </a>
    <a
      v-if="showPlayStoreLink"
      class="flex-container items-center"
      :href="PLAY_STORE_URL"
      target="_blank"
      noreferrer
      noopener
      data-test-id="play-store-link"
    >
      <IconComponent icon="googlePlay" />
      {{ $t('general.playStore') }}
    </a>
  </footer>
</template>
<style scoped>
footer {
  @apply uppercase text-xs;
  @apply flex-container gap-4 justify-center;
}
</style>
