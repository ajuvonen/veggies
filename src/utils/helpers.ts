import {useMemoize} from '@vueuse/core';
import type {ChartOptions, ChartType, Scale} from 'chart.js';
import type {Context} from 'chartjs-plugin-datalabels';
import {DateTime} from 'luxon';
import {mergeDeep, sample} from 'remeda';
import {
  ALL_VEGGIES,
  BEANS,
  CATEGORY_EMOJI,
  COLORS,
  FRUITS,
  GRAINS,
  LEAFIES,
  MUSHROOMS,
  ROOTS,
  VEGETABLES,
} from '@/utils/constants';
import {AchievementLevel, Category} from '@/utils/types';

export const getCategoryForVeggie = useMemoize((veggie: string) => {
  if (FRUITS.includes(veggie)) {
    return Category.Fruit;
  } else if (VEGETABLES.includes(veggie)) {
    return Category.Vegetable;
  } else if (LEAFIES.includes(veggie)) {
    return Category.Leafy;
  } else if (ROOTS.includes(veggie)) {
    return Category.Root;
  } else if (BEANS.includes(veggie)) {
    return Category.Bean;
  } else if (GRAINS.includes(veggie)) {
    return Category.Grain;
  } else if (MUSHROOMS.includes(veggie)) {
    return Category.Mushroom;
  }
  return undefined;
});

export const getChartOptions = <T extends ChartType>(
  grids: boolean,
  stacked: boolean,
  withIcons: boolean,
  showChartAnimations: boolean,
  overrides: Partial<ChartOptions<T>> = {},
) =>
  mergeDeep(
    {
      animation: !showChartAnimations ? false : undefined,
      responsive: true,
      maintainAspectRatio: !grids,
      normalized: true,
      layout: {
        padding: 0,
      },
      scales: grids
        ? {
            y: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
            y1: {
              position: 'right',
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
              afterBuildTicks: (axis: Scale) => {
                axis.ticks = [...axis.chart.scales.y.ticks];
                axis.min = axis.chart.scales.y.min;
                axis.max = axis.chart.scales.y.max;
              },
            },
            x: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
          }
        : undefined,
      plugins: {
        title: {
          display: false,
        },
        legend: {
          display: false,
        },
        datalabels: {
          ...(withIcons
            ? {
                anchor: 'center',
                align: 'center',
                font: {
                  size: 25,
                },
                textShadowColor: '#fff',
                textShadowBlur: 5,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                formatter: (value: any, {dataset: {label}}: Context) =>
                  value ? CATEGORY_EMOJI[label as Category] : '',
              }
            : {
                display: false,
              }),
        },
        tooltip: {
          animation: showChartAnimations,
          padding: 8,
          titleFont: {
            size: 14,
            weight: 'normal',
          },
          bodyFont: {
            size: 14,
          },
          footerFont: {
            size: 14,
            weight: 'normal',
          },
          displayColors: false,
          backgroundColor: COLORS.darkBlue,
          bodyColor: COLORS.offWhite,
          titleColor: COLORS.offWhite,
        },
      },
    },
    overrides,
  ) as ChartOptions<T>;

export const getRandomVeggie = () => ALL_VEGGIES[Math.floor(Math.random() * ALL_VEGGIES.length)];

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateParser = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return DateTime.fromISO(value.split('T')[0]);
  }
  return value;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateReplacer = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return value.split('T')[0];
  }
  return value;
};

const veggieEmojis: readonly string[] = [
  '🥝',
  '🥥',
  '🍇',
  '🍈',
  '🍉',
  '🍊',
  '🍋',
  '🍌',
  '🍍',
  '🥭',
  '🍎',
  '🍏',
  '🍐',
  '🍑',
  '🍒',
  '🍓',
  '🫐',
  '🍅',
  '🍆',
  '🌽',
  '🌶️',
  '🫑',
  '🥑',
  '🥒',
  '🥬',
  '🥦',
  '🥔',
  '🧄',
  '🧅',
  '🥕',
  '🫛',
  '🥜',
  '🫘',
  '🌿',
  '🍄‍🟫',
  '🌾',
  '🌰',
];
export const getRandomEmojis = (amount: number = 1) => sample(veggieEmojis, amount);

export const achievementLevelHelper = (levels: [number, AchievementLevel][], value: number) =>
  levels.find(([threshold]) => value >= threshold)?.[1] ?? AchievementLevel.NoAchievement;
