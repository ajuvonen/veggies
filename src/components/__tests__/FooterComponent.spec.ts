import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import {BLUESKY_URL, PLAY_STORE_URL} from '@/utils/constants';
import FooterComponent from '@/components/FooterComponent.vue';

describe('FooterComponent', () => {
  it('renders', () => {
    const wrapper = mount(FooterComponent);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findByTestId('bluesky-link').attributes('href')).toBe(BLUESKY_URL);
  });

  it('renders with play store link', async () => {
    const originalNavigator = navigator.userAgent;
    Object.defineProperty(navigator, 'userAgent', {
      value:
        'Mozilla/5.0 (Linux; Android 14; SAMSUNG SM-G973U) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/14.2 Chrome/87.0.4280.141 Mobile Safari/537.36',
      configurable: true,
    });
    const wrapper = mount(FooterComponent);
    try {
      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.findByTestId('bluesky-link').exists()).toBe(true);
      expect(wrapper.findByTestId('play-store-link').exists()).toBe(true);
      expect(wrapper.findByTestId('play-store-link').attributes('href')).toBe(PLAY_STORE_URL);
    } finally {
      Object.defineProperty(navigator, 'userAgent', {
        value: originalNavigator,
        configurable: true,
      });
    }
  });

  it('does not render play store link in webview', async () => {
    const originalNavigator = navigator.userAgent;
    Object.defineProperty(navigator, 'userAgent', {
      value:
        'Mozilla/5.0 (Linux; Android 14; SAMSUNG SM-G973U) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/14.2 Chrome/87.0.4280.141 Mobile Safari/537.36 **wv**',
      configurable: true,
    });
    const wrapper = mount(FooterComponent);
    try {
      expect(wrapper.findByTestId('bluesky-link').exists()).toBe(true);
      expect(wrapper.findByTestId('play-store-link').exists()).toBe(false);
    } finally {
      Object.defineProperty(navigator, 'userAgent', {
        value: originalNavigator,
        configurable: true,
      });
    }
  });
});
