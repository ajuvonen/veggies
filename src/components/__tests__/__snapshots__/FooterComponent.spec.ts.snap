// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterComponent > renders 1`] = `
"<footer data-v-7490b180=""><a data-v-7490b180="" class="flex-container items-center" href="https://bsky.app/profile/eatyourveggies.app" target="_blank" noreferrer="" noopener="" data-test-id="bluesky-link"><svg data-v-7490b180="" style="width: 1.25rem; height: 1.25rem;" class="shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M12 10.8c-1.087-2.114-4.046-6.053-6.798-7.995C2.566.944 1.561 1.266.902 1.565.139 1.908 0 3.08 0 3.768c0 .69.378 5.65.624 6.479.815 2.736 3.713 3.66 6.383 3.364.136-.02.275-.039.415-.056-.138.022-.276.04-.415.056-3.912.58-7.387 2.005-2.83 7.078 5.013 5.19 6.87-1.113 7.823-4.308.953 3.195 2.05 9.271 7.733 4.308 4.267-4.308 1.172-6.498-2.74-7.078a8.741 8.741 0 0 1-.415-.056c.14.017.279.036.415.056 2.67.297 5.568-.628 6.383-3.364.246-.828.624-5.79.624-6.478 0-.69-.139-1.861-.902-2.206-.659-.298-1.664-.62-4.3 1.24C16.046 4.748 13.087 8.687 12 10.8Z" class="transition-all"></path>
    </svg> Follow on Bluesky</a>
  <!--v-if-->
</footer>"
`;

exports[`FooterComponent > renders with play store link 1`] = `
"<footer data-v-7490b180=""><a data-v-7490b180="" class="flex-container items-center" href="https://bsky.app/profile/eatyourveggies.app" target="_blank" noreferrer="" noopener="" data-test-id="bluesky-link"><svg data-v-7490b180="" style="width: 1.25rem; height: 1.25rem;" class="shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M12 10.8c-1.087-2.114-4.046-6.053-6.798-7.995C2.566.944 1.561 1.266.902 1.565.139 1.908 0 3.08 0 3.768c0 .69.378 5.65.624 6.479.815 2.736 3.713 3.66 6.383 3.364.136-.02.275-.039.415-.056-.138.022-.276.04-.415.056-3.912.58-7.387 2.005-2.83 7.078 5.013 5.19 6.87-1.113 7.823-4.308.953 3.195 2.05 9.271 7.733 4.308 4.267-4.308 1.172-6.498-2.74-7.078a8.741 8.741 0 0 1-.415-.056c.14.017.279.036.415.056 2.67.297 5.568-.628 6.383-3.364.246-.828.624-5.79.624-6.478 0-.69-.139-1.861-.902-2.206-.659-.298-1.664-.62-4.3 1.24C16.046 4.748 13.087 8.687 12 10.8Z" class="transition-all"></path>
    </svg> Follow on Bluesky</a><a data-v-7490b180="" class="flex-container items-center" href="https://play.google.com/store/apps/details?id=app.eatyourveggies.twa" target="_blank" noreferrer="" noopener="" data-test-id="play-store-link"><svg data-v-7490b180="" style="width: 1.25rem; height: 1.25rem;" class="shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M22.018 13.298l-3.919 2.218-3.515-3.493 3.543-3.521 3.891 2.202a1.49 1.49 0 0 1 0 2.594zM1.337.924a1.486 1.486 0 0 0-.112.568v21.017c0 .217.045.419.124.6l11.155-11.087L1.337.924zm12.207 10.065l3.258-3.238L3.45.195a1.466 1.466 0 0 0-.946-.179l11.04 10.973zm0 2.067l-11 10.933c.298.036.612-.016.906-.183l13.324-7.54-3.23-3.21z" class="transition-all"></path>
    </svg> Google Play</a></footer>"
`;
